x/wasm/binding/cw_struct/cw_query.go

```diff
@@ -0,0 +1,210 @@
+package cw_struct
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/NibiruChain/nibiru/x/common/asset"
+	perpammtypes "github.com/NibiruChain/nibiru/x/perp/amm/types"
+)
+
+// BindingQuery corresponds to the BindingQuery enum in CosmWasm binding
+// contracts (Rust). It specifies which queries can be called into the
+// Nibiru bindings, and describes their JSON schema for connecting app ⇔ Wasm.
+//
+// ### Note
+// 1. The JSON field names must match the ones on the smart contract
+type BindingQuery struct {
+	Reserves        *ReservesRequest        `json:"reserves,omitempty"`
```

Comment:
Same here with the pointers, is this needed?

Reply:
Both `BindingQuery` and `BindingMsg` are basically stand-ins for Rust `enum` types, which work a bit differently than the `iota` ones in Golang. Essentially, we need a type that parses 1-to-1 with [NibiruQuery](https://github.com/NibiruChain/cw-nibiru/blob/90df123f8d32d47b5b280ec6ae7dde0f9dbf2787/contracts/bindings-perp/src/query.rs) and [ExecuteMsg](https://github.com/NibiruChain/cw-nibiru/blob/90df123f8d32d47b5b280ec6ae7dde0f9dbf2787/contracts/bindings-perp/src/msg.rs)

Reply:
I'll add a comment about this on the struct.